### ISSUE TITLE
Image Intensity Rescaling in Preprocessing

### ISSUE THREAD

**Issue by MAeolus** [alibaba-damo-academy/self-supervised-anatomical-embedding-v2#8]
Thanks for your brilliant work!
I have some questions about the implementation.

1. 

According to the configuration files and preprocessing code, the input image intensity is rescaled to the range (0, 255) with an additional bias of 50. `dict(type='RescaleIntensity', out_min_max=(0, 255), in_min_max=(0, 1), bias=50),` This can be observed in different experiments settings, for example, [samv2_ct_mri_intra.py](https://github.com/alibaba-damo-academy/self-supervised-anatomical-embedding-v2/blob/035bbfc1816e40c122303a5e21accbdc35beac43/configs/samv2/samv2_ct_mri_intra.py#L94) and [samv2_ct_mri_inter.py](https://github.com/alibaba-damo-academy/self-supervised-anatomical-embedding-v2/blob/035bbfc1816e40c122303a5e21accbdc35beac43/configs/samv2/samv2_ct_mri_inter.py#L121).

I would like to confirm if this rescaling and bias are intentional design choices and why this approach is applied instead of simply normalizing the intensity to (0, 1). Additionally, I would like to know if the released pretrained models were trained using this setting?

2.

The paper does not specify which embedding was used in each experiment. Were all the results obtained by calculating the similarity score matrices separately for the three embeddings—fine appearance, coarse appearance, and semantic—and then averaging them?

 Appreciate your response.

**Comment by aa1234241** [comment]
Hello, thanks for your interest.

R1: I first rescale the intensity to the range [0, 1] because the augmentations, such as RandomNoise3d and RandomBlur3d, require inputs within this range. Afterward, I rescale back to [0, 255], as the pretrained ResNet-18 model I use expects this setting. Removing this second rescaling step also works, but the pretrained models typically operate with this additional rescaling.
R2:[get_sim_embed_semantic_loc](https://github.com/alibaba-damo-academy/self-supervised-anatomical-embedding-v2/blob/035bbfc1816e40c122303a5e21accbdc35beac43/tools/interfaces.py#L191) Yes, I first calculate the similarity map for each embedding, then interpolate the three similarity maps into the same size and find the maximum value location.

**Comment by sifaoso** [comment]
Hey @aa1234241! 
I'm also trying to reproduce your results on NIH-LN dataset (for both SAM and UAE-S), and I encounter the same issue regarding intensity rescaling. In fact, the current code uses the `RescaleIntensity` class which rescales input data to [-50, 200] range (i.e., it first rescales to [0, 255], and then substract a bias of 50). It does not rescale the intensity to the [0, 1] range, and hence the `RandomNoise3d` augmentation has no impact on the 3D patch, since the `std=(0, 001)`. Any help on this please?

cf. [sam/datasets/piplines/process3d.py](https://github.com/alibaba-damo-academy/self-supervised-anatomical-embedding-v2/blob/035bbfc1816e40c122303a5e21accbdc35beac43/sam/datasets/piplines/process3d.py#L953)

**Comment by aa1234241** [comment]
> Hey [@aa1234241](https://github.com/aa1234241)! I'm also trying to reproduce your results on NIH-LN dataset (for both SAM and UAE-S), and I encounter the same issue regarding intensity rescaling. In fact, the current code uses the `RescaleIntensity` class which rescales input data to [-50, 200] range (i.e., it first rescales to [0, 255], and then substract a bias of 50). It does not rescale the intensity to the [0, 1] range, and hence the `RandomNoise3d` augmentation has no impact on the 3D patch, since the `std=(0, 001)`. Any help on this please?
> 
> cf. [sam/datasets/piplines/process3d.py](https://github.com/alibaba-damo-academy/self-supervised-anatomical-embedding-v2/blob/035bbfc1816e40c122303a5e21accbdc35beac43/sam/datasets/piplines/process3d.py#L953)

Thanks for catching this! You can simply remove the random noise augmentation or adjust it to the correct scale. It shouldn’t affect the final embeddings significantly.
